### ISSUE TITLE
Link search does not prioritise Posts

### ISSUE THREAD

**Issue by getdave** [WordPress/gutenberg#67561]
For general Link searches which haven't been preconfigured to respect a particular type then Posts should always be favoured in the results above other types such as Taxonomy or Media...etc

------

> Since @getdave mentioned "We could look into this more if there is sufficient interest on this Issue." I wanted to note that [gizmo2501](https://wordpress.org/support/users/gizmo2501/) is experiencing issues with linking that could be resolved by the ability to customize results: [Linking is prioritising attachments instead of posts](https://wordpress.org/support/topic/linking-is-prioritising-attachments-instead-of-posts/) 

 _Originally posted by @eidolonnight in [#31496](https://github.com/WordPress/gutenberg/issues/31496#issuecomment-2515534289)_

**Comment by getdave** [comment]
https://github.com/WordPress/gutenberg/pull/67563 is experimenting with weighting slightly towards Posts in the results.

The Playground link should appear here soon which will allow anyone to test this PR.

**Comment by shuttermuse** [comment]
Is this why, from 6.7.1 onwards, my link searches now show attachments before posts?  There was a marked and annoying change going from 6.7 to 6.7.1 that I can't see overtly mentioned in the changelog.

When I first noticed it, I rolled back to 6.7 to check, and the issue disappeared. Then I moved to 6.7.1, and it came back.

Search results used to prioritize posts, which was ideal for internal linking. Now, I get attachments at the top, and in 20 years of blogging, I have never needed to link text to an attachment.
